src/Source/ChainSourceIterator.php

```diff
@@ -16,10 +16,13 @@
 final class ChainSourceIterator implements SourceIteratorInterface
 {
     /**
-     * @var \ArrayIterator
+     * @var \ArrayIterator<array-key, SourceIteratorInterface>
      */
     private $sources;
 
+    /**
+     * @param array<SourceIteratorInterface> $sources
```

Comment:
Do you know if there is a php fixer rule to avoid having both: Foo[] and array<Foo> in the codebase?

Reply:
I would say there is none